src/server/transaction.cc

```diff
@@ -484,6 +484,8 @@ bool Transaction::RunInShard(EngineShard* shard, bool txq_ooo) {
       local_result_ = status;
     } else {
       if (status == OpStatus::OUT_OF_MEMORY) {
+        absl::base_internal::SpinLockHolder lk{&local_result_mu_};
+        CHECK(local_result_ == OpStatus::OK || local_result_ == OpStatus::OUT_OF_MEMORY);
```

Comment:
I  believe we have the same issue below inside catch

Reply:
True, fixed